src/main/java/org/wildfly/transaction/client/provider/jboss/JBossLocalTransactionProvider.java

```
@@ -361,7 +361,7 @@ public void rollback() throws XAException {
             int oldVal;
             do {
                 oldVal = completionBits.get();
-                if ((oldVal & BIT_PREPARE_OR_ROLLBACK) != 0) {
+                if ((oldVal & BIT_PREPARE_OR_ROLLBACK) == 0) {
```

In this case we keep the '!=' because we want to throw an exception if rollback() was called after prepare() or rollback().